File: backend/academic-management-hub/src/main/java/com/academichub/academic_management_hub/models/Professor.java
package com.academichub.academic_management_hub.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.GenericGenerator;

import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "professors")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Professor {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @NotNull
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "department_id", nullable = false)
    private Department department;

    @Enumerated(EnumType.STRING)
    @Column(name = "title")
    private ProfessorTitle title;

    @NotBlank
    @Column(name = "first_name", nullable = false)
    private String firstName;

    @NotBlank
    @Column(name = "last_name", nullable = false)
    private String lastName;

    @Column(name = "office_location")
    private String officeLocation;

    @Column(name = "phone")
    private String phone;

    // Store as comma-separated string for testing with H2
    @Column(name = "research_interests")
    private String researchInterests;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;

    // Helper methods for research interests
    public List<String> getResearchInterestsList() {
        if (researchInterests == null || researchInterests.isEmpty()) {
            return new ArrayList<>();
        }
        return List.of(researchInterests.split(","));
    }

    public void setResearchInterestsList(List<String> interests) {
        if (interests == null || interests.isEmpty()) {
            this.researchInterests = "";
            return;
        }
        this.researchInterests = String.join(",", interests);
    }
}

File: frontend/src/types/professor.ts
// src/types/professor.ts

// Match the ProfessorTitle enum from Java
export enum ProfessorTitle {
  ASSISTANT_PROFESSOR = 'ASSISTANT_PROFESSOR',
  ASSOCIATE_PROFESSOR = 'ASSOCIATE_PROFESSOR',
  PROFESSOR = 'PROFESSOR',
  DISTINGUISHED_PROFESSOR = 'DISTINGUISHED_PROFESSOR',
  EMERITUS_PROFESSOR = 'EMERITUS_PROFESSOR',
  VISITING_PROFESSOR = 'VISITING_PROFESSOR',
  ADJUNCT_PROFESSOR = 'ADJUNCT_PROFESSOR'
}

export interface Professor {
  id: string;
  userId: string;  // From the @OneToOne User relationship
  departmentId: string;  // From the @ManyToOne Department relationship
  title: ProfessorTitle;
  firstName: string;
  lastName: string;
  officeLocation?: string;
  phone?: string;
  researchInterests?: string[];  // We'll receive this as an array from the API
  createdAt: string;
  updatedAt: string;
}

// Additional interfaces for specific views or responses
export interface ProfessorWithDepartment extends Professor {
  department: {
    id: string;
    name: string;
  };
}

export interface ProfessorDetails extends Professor {
  email: string;  // From associated user
  isActive: boolean;  // From associated user
  lastLogin?: string;  // From associated user
}